### PR TITLE
Fixed segfault during stop play

### PR DESCRIPTION
**Pull request by MathewMacDougall**
### Description

When running the `StopPlay` once all the robots were more or less at their final destinations, the AI would segfault. It turns out this was because the `MoveAction` in the `GoalieTactic` did not have `loop_forever` set to true, so the `GoalieTactic` would sometimes return a nullptr in certain cases. Furthermore, we were setting MotionConstraints before checking if the returned intent was actually valid, which is what triggered this error.

MotionConstraints are now set _after_ the null check, and the `MoveAction` now has `loop_forever = true`

### Testing Done

Lots of manual testing. STP should be tested better once it's refactored (see #783 ).

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/ai/hl/stp/tactic/goalie_tactic.cpp [hunk 1]
```diff
@@ -131,7 +131,9 @@ std::optional<Point> GoalieTactic::restrainGoalieInRectangle(
 
 void GoalieTactic::calculateNextIntent(IntentCoroutine::push_type &yield)
 {
-    MoveAction move_action = MoveAction();
+    MoveAction move_action =
+        MoveAction(MoveAction::ROBOT_CLOSE_TO_DEST_THRESHOLD,
+                   MoveAction::ROBOT_CLOSE_TO_ORIENTATION_THRESHOLD, true);
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/tactic/goalie_tactic.cpp:136]: Can we just remove the default constructor for MoveAction while we're here?

**Review by jonathanlew — COMMENTED** [review]
Nice find!

**Comment by MathewMacDougall** [comment]
@HaomiaoZ 's changes from #767 should help prevent this in the future by forcing people to be more explicit.

File: src/software/ai/hl/stp/tactic/goalie_tactic.cpp (hunk 1 above)
Comment (MathewMacDougall) [reply src/software/ai/hl/stp/tactic/goalie_tactic.cpp:136]: #767 will take care of that for multiple Actions and tactics.

**Review by jonathanlew — APPROVED** [review]
Looks good!

**Review by akhilveeraghanta — APPROVED** [review]
LGTM that's pretty hectic :shipit:

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Review by EvanMorcom — APPROVED** [review]
:100: No-segfaulto

**Comment by MathewMacDougall** [comment]
Segfaulto begone-o

**Pull request merged**